stdlib/encodings/utf_8_sig.pyi

```diff
@@ -0,0 +1,27 @@
+import codecs
+
+class IncrementalEncoder(codecs.IncrementalEncoder):
+    def __init__(self, errors: str = ...) -> None: ...
+    def encode(self, input: str, final: bool = ...) -> bytes: ...
+    def reset(self) -> None: ...
+    def getstate(self) -> int | str: ...
+    def setstate(self, state: int | str) -> None: ...
```

Comment:
N.B. in the source code this is really:
```python
def getstate(self) -> int: ...
def setstate(self, state: int) -> None: ...
```
However, when I type it like that I get a mypy error pointing me [here](https://mypy.readthedocs.io/en/stable/common_issues.html#incompatible-overrides) and saying that this violates "The Liskov Substitution Principle" so I changed it to match the `codecs.IncrementalEncoder` method and satisfy mypy.

Link to the action with the relevant mypy error [here](https://github.com/python/typeshed/runs/4632887034?check_suite_focus=true)

I think adding the correct typing is highlighting that the source code is violating the LSP rather than it being an issue with the typing.

Opinions on leaving the typing like it is here VS just using `int` for typing and inline ignoring that mypy error?

Reply:
If it violates LSP then that's the "fault" of the people who wrote the code, rather than the fault of the people who are writing the stubs. I think it's generally more important for the stubs to be accurate, so I'd type it like it appears in the source code and add relevant `type: ignore[override]` comments, personally 🙂

N.B. I haven't actually read the source code, this is just based on what you've written!

Reply:
Agreed, this should just be `int` with a `# type: ignore` comment. There are many other similar places in typeshed.

Reply:
Perfect I'll revert that then. Thanks 😄